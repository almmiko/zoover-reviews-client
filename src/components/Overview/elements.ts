import styled from 'styled-components';

const MARGIN = 60;

export const Wrapper = styled.div`
    width: 30%;
`;

export const RatingContainer = styled.div`
  margin-top: ${MARGIN}px;
`;

export const StatsSectionContainer = styled.div`
  margin-top: ${MARGIN}px;
`;

export const TraveledWithContainer = styled.div`
  margin-top: ${MARGIN}px;
`;
