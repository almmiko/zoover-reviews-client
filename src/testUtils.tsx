import React from 'react';

export const reviewStoreStub = {
  fetchReviewComments: jest.fn(async () => {}),
  fetchReviewStats : jest.fn(async () => {})
};


