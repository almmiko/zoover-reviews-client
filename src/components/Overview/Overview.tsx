import React from 'react';
import StatsSection from './components/StatsSection/StatsSection';
import Rating from './components/Rating/Rating';
import { Wrapper, RatingContainer, StatsSectionContainer, TraveledWithContainer } from './elements';
import { inject, observer } from 'mobx-react';
import { get } from 'lodash';
import ReviewsStore from '../../stores/reviewsStore';

type InjectedProps = {
  reviewsStore: ReviewsStore;
}

type Props = {}

@inject('reviewsStore')
@observer
class Overview extends React.Component<Props> {

  get injected(): InjectedProps {
    return this.props as Props & InjectedProps;
  }

  render() {
    const { reviewsStore } = this.injected;
    const { statsLoaded, getReviewStats } = reviewsStore;

    const rating = get(getReviewStats, 'averageRating');
    const ratingAspects = get(getReviewStats, 'averageRatingAspects');
    const traveledWith = get(getReviewStats, 'traveledWith');

    return (
      <Wrapper>
        <RatingContainer>
          <Rating loaded={statsLoaded} rating={rating} />
        </RatingContainer>
        <StatsSectionContainer>
          <StatsSection
            title={'Rating aspects'}
            stats={ratingAspects}
            loaded={statsLoaded} />
        </StatsSectionContainer>
        <TraveledWithContainer>
          <StatsSection
            title={'Traveled with'}
            stats={traveledWith}
            loaded={statsLoaded} />
        </TraveledWithContainer>
      </Wrapper>
    );
  }
}

export default Overview;
