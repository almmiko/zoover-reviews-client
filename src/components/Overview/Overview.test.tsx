import { shallow } from 'enzyme';
import Overview from './Overview';
import React from 'react';
import { reviewStoreStub } from '../../testUtils';

it('renders correctly', () => {
  // @ts-ignore
  const tree = shallow(<Overview.wrappedComponent apiError={false} reviewsStore={reviewStoreStub} />);
  expect(tree).toMatchSnapshot();
});
