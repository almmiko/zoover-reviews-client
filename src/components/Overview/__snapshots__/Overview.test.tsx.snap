// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <Rating
      apiError={false}
    />
  </styled.div>
  <styled.div>
    <StatsSection
      apiError={false}
      title="Rating aspects"
    />
  </styled.div>
  <styled.div>
    <StatsSection
      apiError={false}
      title="Traveled with"
    />
  </styled.div>
</styled.div>
`;
